### DISCUSSION TITLE
[Suggestion] Elevator logic block for Rails and spread barrel for laser

### DISCUSSION THREAD

**Discussion by Dreamseizer36b** [tsunamayo/Starship-EVO#3965] [Suggestions]
[Elevator Logic Block]
The idea is a logic block that makes the rail stop at a certain height.
Say you have 5 floors, you can connect 5 of these to the rail and set the height for each.
To make it go to that exact height you simply connect a different button to each one so when pressed it goes to the height designated by the logic block.

[Spread Barrel for Laser]
Basically a shotgun barrel that shoots multiple projectiles in a cone, damage is increased overall but divided between projectiles.
Each barrel length adds 3 projectiles, so if the maximum length is 4 it'd be 3,6,9,12.

While I'm at it
[Blast Attachment]
Increases projectile size and hull damage at the cost of increased recharge time and energy drain.

**Comment by cloudrambler** [comment]
The "elevator" logic function would be useful in many applications; such as sequencing the movements of complicated landing gear. The game does need some form of multi-input or multi-path sequencer. A similar method was proposed in #3873

**Comment by Garrett-C** [comment]
Yeah I have spent the last day trying to build an elevator for my ship and boy does it suck to add more than 2 floors right now. 

Don't really mind how it's done but some easier way of doing multifloor elevators would be a big plus.

**Comment by tsunamayo** [comment]
@Dreamseizer36b Yes we have a big functionality gap regarding elevator, I will address that in the future. 
Your barrel suggestion are also quite interesting. The spread barrel looks a bit like the gatling, except it would shoot all at one...
Thanks
